resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/InMemoryCircuitBreakerRegistry.java

```diff
@@ -18,74 +18,132 @@
+		this.sharedCircuitBreakerConfiguration = new ConcurrentHashMap<>();
+		this.sharedCircuitBreakerConfiguration.put(DEFAULT_CONFIG, defaultCircuitBreakerConfig);
+	}
+
+	/**
+	 * The constructor with custom default circuitBreaker properties.
+	 *
+	 * @param defaultCircuitBreakerConfig The BackendMonitor service properties.
+	 */
+	public InMemoryCircuitBreakerRegistry(CircuitBreakerConfig defaultCircuitBreakerConfig) {
+		this.defaultCircuitBreakerConfig = Objects.requireNonNull(defaultCircuitBreakerConfig, "CircuitBreakerConfig must not be null");
+		this.circuitBreakers = new ConcurrentHashMap<>();
+		this.postCreationConsumers = new ArrayList<>();
+		this.sharedCircuitBreakerConfiguration = new ConcurrentHashMap<>();
+		this.sharedCircuitBreakerConfiguration.put(DEFAULT_CONFIG, defaultCircuitBreakerConfig);
+	}
+
+	@Override
+	public Seq<CircuitBreaker> getAllCircuitBreakers() {
+		return Array.ofAll(circuitBreakers.values());
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public CircuitBreaker circuitBreaker(String name) {
+		return circuitBreakers.computeIfAbsent(Objects.requireNonNull(name, NAME_MUST_NOT_BE_NULL),
+				k -> postCreateCircuitBreaker(CircuitBreaker.of(name, defaultCircuitBreakerConfig)));
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public CircuitBreaker circuitBreaker(String name, CircuitBreakerConfig customCircuitBreakerConfig) {
+		return circuitBreakers.computeIfAbsent(Objects.requireNonNull(name, NAME_MUST_NOT_BE_NULL),
+				k -> postCreateCircuitBreaker(CircuitBreaker.of(name, customCircuitBreakerConfig)));
+	}
+
+	@Override
+	public CircuitBreaker circuitBreaker(String name, Supplier<CircuitBreakerConfig> circuitBreakerConfigSupplier) {
+		return circuitBreakers.computeIfAbsent(Objects.requireNonNull(name, NAME_MUST_NOT_BE_NULL),
+				k -> {
+					CircuitBreakerConfig config = circuitBreakerConfigSupplier.get();
+					return postCreateCircuitBreaker(CircuitBreaker.of(name, config));
+				});
+	}
+
+	@Override
+	public void addConfiguration(String configName, CircuitBreakerConfig configuration) {
+		if (configName.equals(DEFAULT_CONFIG)) {
+			throw new IllegalArgumentException("you can not use 'default' as a configuration name as it is preserved for default configuration");
+		}
+		this.sharedCircuitBreakerConfiguration.put(configName, configuration);
+	}
+
+	@Override
+	public Optional<CircuitBreakerConfig> getConfigurationByName(String configName) {
+		return Optional.ofNullable(this.sharedCircuitBreakerConfiguration.get(configName));
+	}
+
+	@Override
+	public void registerPostCreationConsumer(Consumer<CircuitBreaker> postCreationConsumer) {
+		postCreationConsumers.add(postCreationConsumer);
+	}
+
+	@Override
+	public void unregisterPostCreationConsumer(Consumer<CircuitBreaker> postCreationConsumer) {
+		postCreationConsumers.remove(postCreationConsumer);
+	}
+
+	private CircuitBreaker postCreateCircuitBreaker(CircuitBreaker createdCircuitBreaker) {
```

Comment:
Could be moved into AbstractRegistry.
Please rename to `notifyPostCreationConsumers`.